enterprise/views.py

```diff
@@ -808,6 +817,68 @@ def post(self, request):
         return redirect(success_url if consent_provided else failure_url)
 
 
+class EnterpriseSelectionView(FormView):
+    """
+    Allow an enterprise learner to activate one of learner's linked enterprises.
+    """
+
+    form_class = EnterpriseSelectionForm
+    template_name = 'enterprise/enterprise_customer_select_form.html'
+
+    @method_decorator(login_required)
+    def dispatch(self, request, *args, **kwargs):
```

Comment:
what is the reason for overriding this `dispatch`?

Reply:
To ensure that only the logged in user can access to page.

Reply:
If that is the case then I think this is more clean way to enforce `login_required`
```
@method_decorator(login_required, name='dispatch')
class EnterpriseSelectionView(FormView):
```